### DISCUSSION TITLE
date time picker value invalid

### DISCUSSION THREAD

**Discussion by yuniit** [element-plus/element-plus#13361] [Q&A] [answered]
Hi guys, I have an issue with datetime picker after changing value in the input  without clicking save it does not return the correct value

https://github.com/element-plus/element-plus/assets/30621352/81fe7f9a-8f45-4597-b838-3903b39dde9a

[playground link](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6Ijx0ZW1wbGF0ZT5cbiAgPGRpdiBjbGFzcz1cImRlbW8tZGF0ZS1waWNrZXJcIj5cbiAgICA8ZGl2IGNsYXNzPVwiYmxvY2tcIj5cbiAgICAgIDxwPkNvbXBvbmVudCB2YWx1Ze+8mnt7IHZhbHVlIH19PC9wPlxuICAgICAgPGVsLWRhdGUtcGlja2VyIHYtbW9kZWw9XCJ2YWx1ZVwiIHR5cGU9XCJkYXRldGltZXJhbmdlXCIgc3RhcnQtcGxhY2Vob2xkZXI9XCJTdGFydCBkYXRlXCIgZW5kLXBsYWNlaG9sZGVyPVwiRW5kIGRhdGVcIiAvPlxuICAgICAgPGJyPlxuICAgICAgPGJyPlxuICAgICAgPGJyPlxuICAgICAgPGVsLWJ1dHRvbiB0eXBlPVwicHJpbWFyeVwiIEBjbGljaz1cInN1Ym1pdFwiPiBTdWJtaXQgPC9lbC1idXR0b24+XG4gICAgPC9kaXY+XG5cbiAgPC9kaXY+XG48L3RlbXBsYXRlPlxuXG48c2NyaXB0IGxhbmc9XCJ0c1wiIHNldHVwPlxuaW1wb3J0IHsgcmVmIH0gZnJvbSAndnVlJ1xuXG5jb25zdCB2YWx1ZSA9IHJlZihbXG4gIG5ldyBEYXRlKCksIG5ldyBEYXRlKCksXG5dKVxuY29uc3QgZGVmYXVsdFRpbWUgPSByZWY8W0RhdGUsIERhdGVdPihbXG4gIG5ldyBEYXRlKDIwMDAsIDEsIDEsIDAsIDAsIDApLFxuICBuZXcgRGF0ZSgyMDAwLCAyLCAxLCAyMywgNTksIDU5KSxcbl0pXG5cbmNvbnN0IHN1Ym1pdCA9ICgpID0+IHtcbiAgY29uc29sZS5sb2coXCJkYXRlOiBcIiwgdmFsdWUudmFsdWUpO1xufTtcblxuPC9zY3JpcHQ+XG48c3R5bGUgc2NvcGVkPlxuLmRlbW8tZGF0ZS1waWNrZXIge1xuICBkaXNwbGF5OiBmbGV4O1xuICB3aWR0aDogMTAwJTtcbiAgcGFkZGluZzogMDtcbiAgZmxleC13cmFwOiB3cmFwO1xufVxuXG4uZGVtby1kYXRlLXBpY2tlciAuYmxvY2sge1xuICBwYWRkaW5nOiAzMHB4IDA7XG4gIHRleHQtYWxpZ246IGNlbnRlcjtcbiAgYm9yZGVyLXJpZ2h0OiBzb2xpZCAxcHggdmFyKC0tZWwtYm9yZGVyLWNvbG9yKTtcbiAgZmxleDogMTtcbn1cblxuLmRlbW8tZGF0ZS1waWNrZXIgLmJsb2NrOmxhc3QtY2hpbGQge1xuICBib3JkZXItcmlnaHQ6IG5vbmU7XG59XG48L3N0eWxlPlxuIiwiaW1wb3J0LW1hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge31cbn0iLCJ0c2NvbmZpZy5qc29uIjoie1xuICBcImNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJhbGxvd0pzXCI6IHRydWUsXG4gICAgXCJjaGVja0pzXCI6IHRydWUsXG4gICAgXCJqc3hcIjogXCJwcmVzZXJ2ZVwiLFxuICAgIFwidGFyZ2V0XCI6IFwiRVNOZXh0XCIsXG4gICAgXCJtb2R1bGVcIjogXCJFU05leHRcIixcbiAgICBcIm1vZHVsZVJlc29sdXRpb25cIjogXCJCdW5kbGVyXCIsXG4gICAgXCJhbGxvd0ltcG9ydGluZ1RzRXh0ZW5zaW9uc1wiOiB0cnVlXG4gIH0sXG4gIFwidnVlQ29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiAzLjNcbiAgfVxufVxuIiwiX28iOnt9fQ==)

**Comment by markhermano** [comment, marked as answer]
@yuniit the dateTimeRange does not apply the changes you made in the range without clicking on the `OK` button but I have a workaround for you.

Use the event `@calendar-change` to get the selected date range and use `@blur` to apply it to your value.
Example:

```html
 <el-date-picker
  v-model="value"
  type="datetimerange"
  start-placeholder="Start date"
  end-placeholder="End date"
  @calendar-change="changeDate"
  @blur="value = selection dates"
 />
 ```
 
Then in your script:
```js
import { ref } from 'vue'

const value = ref([
  new Date(), new Date(),
])
const selectionDates = ref([])

const defaultTime = ref<[Date, Date]>([
  new Date(2000, 1, 1, 0, 0, 0),
  new Date(2000, 2, 1, 23, 59, 59),
])

const changeDate = (data) => {
  selectionDates.value = data
}

const submit = () => {
  console.log("date: ", value.value);
};
```

This is not the most clean workaround but you can start from here. Also, you might want to convert the date format returned by `@calendar-change`.

**Reply by yuniit** [reply]
thank you for answer, that looks good

**Reply by markhermano** [reply]
You're welcome. Hopefully, this answers your question.
